tee-worker/ts-tests/integration-tests/common/transactions.ts

```diff
@@ -217,3 +218,51 @@ export async function multiAccountTxSender(
     expect(eventsResp.length).to.be.equal(txs.length);
     return eventsResp;
 }
+
+// for DI-test
+export const subscribeToEventsWithExtHash = async (
+    requestIdentifier: string,
+    context: IntegrationTestContext
+): Promise<FrameSystemEventRecord[]> => {
+    return new Promise<FrameSystemEventRecord[]>((resolve, reject) => {
+        let blocksToScan = 30;
+
+        const unsubscribe = context.api.rpc.chain.subscribeNewHeads(async (blockHeader) => {
```

Comment:
Maybe we should add a warning to our future selves here to _not_ `await unsubscribe` here :D

Reply:
plz check that I've commented correctly, as I understand it

Reply:
Looks OK; just suggested one amendment above :)